image/descriptor.go

```diff
@@ -69,31 +52,39 @@ func listReferences(w walker) (map[string]*descriptor, error) {
 	return refs, nil
 }
 
-func findDescriptor(w walker, name string) (*descriptor, error) {
-	var d descriptor
-	dpath := filepath.Join("refs", name)
+func findDescriptor(w walker, name string) (*v1.Descriptor, error) {
+	var d v1.Descriptor
+	var index v1.ImageIndex
 
 	switch err := w.walk(func(path string, info os.FileInfo, r io.Reader) error {
-		if info.IsDir() || filepath.Clean(path) != dpath {
+		if info.IsDir() || filepath.Clean(path) != "index.json" {
 			return nil
 		}
 
-		if err := json.NewDecoder(r).Decode(&d); err != nil {
+		if err := json.NewDecoder(r).Decode(&index); err != nil {
 			return err
 		}
 
-		return errEOW
+		for i := 0; i < len(index.Manifests); i++ {
+			if index.Manifests[i].Descriptor.Annotations["org.opencontainers.ref.name"] == name {
+				d = index.Manifests[i].Descriptor
+				return errEOW
+			}
+		}
+
+		return nil
 	}); err {
 	case nil:
-		return nil, fmt.Errorf("%s: descriptor not found", dpath)
+		return nil, fmt.Errorf("index.json: descriptor not found")
 	case errEOW:
 		return &d, nil
 	default:
 		return nil, err
 	}
 }
 
-func (d *descriptor) validate(w walker, mts []string) error {
+//ValidateDescriptor validate descriptor type
+func ValidateDescriptor(d *v1.Descriptor, w walker, mts []string) error {
```

Comment:
No need to export this. It is use unexported interfaces, so it is pointless, anyways.